src/import/kv_importer.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs;
+use std::fmt;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use uuid::Uuid;
+use kvproto::importpb::*;
+
+use config::DbConfig;
+use util::collections::HashMap;
+
+use super::engine::*;
+use super::{Config, Error, Result};
+
+pub type Token = usize;
+
+struct Inner {
+    engines: HashMap<Uuid, Arc<EngineFile>>,
+    clients: HashMap<Token, Arc<EngineFile>>,
+}
+
+pub struct KVImporter {
+    dir: EngineDir,
+    token: AtomicUsize,
```

Comment:
I suggest to rename to a more obvious name, i.e. clientId

Reply:
We use this token name in different places in TiKV, so I think it is fine.

Reply:
Ok